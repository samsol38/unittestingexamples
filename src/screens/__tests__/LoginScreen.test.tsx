import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import LoginScreen from '../login';

describe('LoginScreen render with snapshot', () => {
  test('matches snapshot', () => {
    const tree = render(<LoginScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('LoginScreen events', () => {
  test('renders input fields and login button', () => {
    const {getByPlaceholderText, getByText} = render(<LoginScreen />);
    const emailInput = getByPlaceholderText('Email');
    const passwordInput = getByPlaceholderText('Password');
    const loginButton = getByText('Login');
    expect(emailInput).toBeDefined();
    expect(passwordInput).toBeDefined();
    expect(loginButton).toBeDefined();
  });

  test('updates email state on input change', () => {
    const {getByPlaceholderText} = render(<LoginScreen />);
    const emailInput = getByPlaceholderText('Email');
    fireEvent.changeText(emailInput, 'user@example.com');
    expect(emailInput.props.value).toBe('user@example.com');
  });

  test('updates password state on input change', () => {
    const {getByPlaceholderText} = render(<LoginScreen />);
    const passwordInput = getByPlaceholderText('Password');
    fireEvent.changeText(passwordInput, 'password');
    expect(passwordInput.props.value).toBe('password');
  });

  test('displays error message for invalid credentials', () => {
    const {getByPlaceholderText, getByText} = render(<LoginScreen />);
    const emailInput = getByPlaceholderText('Email');
    const passwordInput = getByPlaceholderText('Password');
    const loginButton = getByText('Login');
    fireEvent.changeText(emailInput, 'invalid@example.com');
    fireEvent.changeText(passwordInput, 'wrongpassword');
    fireEvent.press(loginButton);
    const errorText = getByText('Invalid email or password');
    expect(errorText).toBeDefined();
  });

  test('clears error message after successful login', () => {
    const {getByPlaceholderText, getByText, queryByText} = render(
      <LoginScreen />,
    );
    const emailInput = getByPlaceholderText('Email');
    const passwordInput = getByPlaceholderText('Password');
    const loginButton = getByText('Login');
    fireEvent.changeText(emailInput, 'user@example.com');
    fireEvent.changeText(passwordInput, 'password');
    fireEvent.press(loginButton);
    const errorText = queryByText('Invalid email or password');
    expect(errorText).toBeNull();
  });
});
