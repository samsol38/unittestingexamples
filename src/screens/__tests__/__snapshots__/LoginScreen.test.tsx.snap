// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen render with snapshot matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Email"
    style={
      {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
